test/specs/redos/quadratic_lists.cjs

```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  markdown: '- a\n'.repeat(10000),
+  html: `<ul>${'<li>a</li>'.repeat(10000)}</ul>`
+};
```

Comment:
I can confirm this speeds up parsing long lists.

This test takes about 2 seconds on my machine on master and about 100 ms with this PR.

Reply:
Hooray!